libweb3jsonrpc/Debug.cpp

```diff
@@ -126,6 +126,43 @@ Json::Value Debug::debug_traceBlockByNumber(int _blockNumber, Json::Value const&
 	return ret;
 }
 
+Json::Value Debug::debug_accountRangeAt(
+    string const& _blockHashOrNumber, int _txIndex, string const& _addressHash, int _maxResults)
+{
+    Json::Value ret(Json::objectValue);
+
+    if (_txIndex < 0)
+        throw jsonrpc::JsonRpcException("Negative index");
+    if (_maxResults <= 0)
+        throw jsonrpc::JsonRpcException("Nonpositive maxResults");
+
+    try
+    {
+        Block block = m_eth.block(blockHash(_blockHashOrNumber));
+        size_t const i = std::min(static_cast<size_t>(_txIndex), block.pending().size());
+        State state(State::Null);
+        if (block.info().number() >= m_eth.chainParams().byzantiumForkBlock)
```

Comment:
Shouldn't this be the other way around? We have to create intermediate state in Byzantium+, right?

Reply:
`createIntermediateState` works for both cases (creates state from intermediate root when available, executes previous transactions when not)
Then `State::addresses()` doesn't work right for this state with re-executed transactions (the result of execution is in cache and it doesn't take into account cache)

This `if` shouldn't be here and `State::addresses()` should be fixed, I'm going to do it